scripts/create-component/plop-templates-component/src/components/{{componentName}}/use{{componentName}}.ts.hbs

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps } from '@fluentui/react-compose/lib/next/index';
+import { {{componentName}}Props, {{componentName}}State } from './{{componentName}}.types';
+import { use{{componentName}}State } from './use{{componentName}}State';
+import { useMergedRefs } from '@fluentui/react-hooks';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const {{camelCase componentName}}ShorthandProps = ['loader', 'content'];
+
+const mergeProps = makeMergeProps({ deepMerge: {{camelCase componentName}}ShorthandProps });
+
+/**
+ * Given user props, returns state and render function for a {{componentName}}.
+ */
+export const use{{componentName}} = (props: {{componentName}}Props, ref: React.Ref<HTMLElement>, defaultProps?: {{componentName}}Props) => {
+  // Ensure that the `ref` prop can be used by other things (like useFocusRects) to refer to the root.
+  // NOTE: We are assuming refs should not mutate to undefined. Either they are passed or not.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const resolvedRef = useMergedRefs(ref, React.useRef<HTMLElement>(null));
```

Comment:
I'm pretty sure you can remove the eslint-disable now.

Also since the locally created ref truly is only a backup, instead of merging I still think it would be better to create it on a separate line and only use it when needed. But also not essential.
```ts
const defaultRef = React.useRef<HTMLElement>(null);
const resolvedRef = ref || defaultRef;
const resolvedRef = 
```

Reply:
Updated